### DISCUSSION TITLE
Spindle RPM display on UGS

### DISCUSSION THREAD

**Discussion by Junglesponge** [winder/Universal-G-Code-Sender#2291] [Help] [answered]
Hey everyone, first time posting on here. I hope that I haven't stirred up a common issue here as I did look in the discussions but couldn't see anything similar.

So I am doing testing on my UGS as I am about to install a VFD for my 3 phase spindle. I am looking to be able to control the spindle rpms as desired by the g code or by manual adjustment on the UGS software. There has been many videos and google finds on connections for doing this but as I am waiting for the VFD to get here I am just checking all my ducks are in a row with some testing on the software first. I am using a Arduino and cnc shield v3 controller with grbl 1.1. 

The question or issue I have is that when I type in my m3 s1000 up to m3 s18000 commands  I am getting no rpm response on the UGS screen. When I test the voltage for my spindle enable pin I see it the voltage go from 0 to 4.9v and back again when switched off. I have checked the spindle speeds in the config $$ and adjusted accordingly the max spindle speed. When I move my axis I can see the feed speeds on UGS screen work well and those I also modified in the config $$.

I have watched videos of others with a similar system and there rpms change on screen so wondering where I'm going wrong. Is it cause I have not connected my VFD yet?

![Capture](https://github.com/winder/Universal-G-Code-Sender/assets/142695824/9b140cbb-4136-4c65-9346-a4d71042b865)
![1](https://github.com/winder/Universal-G-Code-Sender/assets/142695824/631aa72a-4dab-476b-8cf0-f6b0b8e1d0f2)
![2](https://github.com/winder/Universal-G-Code-Sender/assets/142695824/2b6b4072-3219-4bc7-b80a-6ced9645c868)

**Comment by Junglesponge** [comment, marked as answer]
Hey thanks for the help guys, good feedback. Luckily was able to sort it.

**Reply by breiler** [reply]
I am glad that you figured it out. But I can't help feeling a tone of sarcasm in your post and I am sorry no one responded to it (me myself just forgot about it). 

It could be good for you to know (for any future questions to this or other open source projects) that we are actually not obligated or paid to support anyone - we do it out of good will in our free time for free.

Out of curiosity (and for the next person having the same issue) how did you fix this? Was your controller configured in laser mode, making the spindle only turning on when G1-movement commands were issued?
